### PR TITLE
chore(demo): add a representations selector to the demo page

### PR DESCRIPTION
**Pull request by brandonocasey**
## Description
While looking into issues related to codec switching and representation/playlist changes I implemented local changes to select representations using the demo ui. I think this would benefit all of us, so I cleaned it up and wrapped it into a pull request.

### CODE REVIEW

File: scripts/index-demo-page.js [hunk 1]
```diff
@@ -277,10 +327,15 @@
         } else {
           sources.dispatchEvent(newEvent('change'));
         }
-        player.on('loadedmetadata', () => {
+        player.on('loadedmetadata', function() {
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:330]: As you can guess, ie 11 didn't like this

File: scripts/index-demo-page.js [hunk 2]
```diff
@@ -165,6 +185,36 @@
     onload();
   };
 
+  var regenerateRepresentations = function() {
+    while (representationsEl.firstChild) {
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:189]: As things are blacklisted representations can be removed/added.

File: scripts/index-demo-page.js [hunk 3]
```diff
@@ -165,6 +185,36 @@
...
+      });
+
+      if (window.mpc.media().id === rep.id) {
+        selectedIndex = i;
+      }
+
+      representationsEl.appendChild(option);
+    });
+
+    representationsEl.selectedIndex = selectedIndex;
```
Comment (brandonocasey) [inline scripts/index-demo-page.js:215]: This does not trigger a `change` event on `representationsEl` as that only happens when not using the js api.

**Comment by gkatsev** [comment]
slowly we're adding stats page features

**Comment by brandonocasey** [comment]
haha yep

**Pull request merged**